superduper/base/metadata.py

```diff
@@ -290,20 +325,20 @@ def get_schema(self, table: str):
         if table in metaclasses:
             return metaclasses[table].class_schema
 
-        r = self.db['Table'].get(identifier=table)
-        try:
-            r = r.unpack()
-            if r['path'] is not None:
-                return import_object(r['path']).class_schema
-            return Schema.build(**r['fields'])
-        except AttributeError as e:
-            if 'unpack' in str(e) and 'NoneType' in str(e):
-                raise NonExistentMetadataError(f'{table} does not exist in metadata')
-            raise e
-        except TypeError as e:
-            if 'NoneType' in str(e):
-                raise NonExistentMetadataError(f'{table} does not exist in metadata')
-            raise e
+        if table in self._schema_cache:
+            return self._schema_cache[table]
+
+        r = self.get_component('Table', table)
+        fields = r['fields']
+
+        # TODO this seems to be to do with json_native
+        if isinstance(fields, str):
+            import json
```

Comment:
In general, try to put all the dependencies at the top-level. 

This makes it easier for tools to detect if some of the dependencies is missing

Reply:
Great